Zend/zend_compile.c

```diff
@@ -7922,12 +7932,14 @@ static void zend_compile_class_decl(znode *result, zend_ast *ast, bool toplevel)
 				zend_build_properties_info_table(ce);
 				ce->ce_flags |= ZEND_ACC_LINKED;
 				zend_observer_class_linked_notify(ce, lcname);
+				zend_inheritance_check_override(ce);
```

Comment:
This also needs an exemption for traits.
```c
if (!(ce->ce_flags & ZEND_ACC_TRAIT)) {
	zend_inheritance_check_override(ce);
}
```
This attribute should only be checked on the method of the class.

Test case:
```php
interface IFace {
    function test();
}

trait Base {
    #[Override]
    function test() {}
}

class Test implements IFace {
    use Base;
}
```

Reply:
> Adding this check (along with a ce->parent_name == NULL) to zend_compile_prop_decl (where you add ZEND_ACC_OVERRIDE) could avoid looping the properties again, and would also reduce the risk of missing this check somewhere for classes that don't do inheritance.

I'm not sure if I follow. `zend_compile_prop_decl` does not appear appropriate, because that is about properties, not methods.

It's likely that there is a better place for the check, but one thing that might or might not make it more complicated is that we can only check whether the method is valid or not, one we validated the parent class and all interfaces. Only if *none* of them have the method, the error must be emitted. This is different to the method variance checks, for example, where we only need to find a mismatching signature and can immediately error, because “later” interfaces cannot fix that.

Reply:
Let's keep it the way it is. It's consistently at the site of `ZEND_ACC_LINKED`.